### PR TITLE
Import: improve the error message when user tries to upload a wpress file

### PR DESCRIPTION
**Pull request by jimmy319**
#### Proposed Changes

* To improve the user experience, make the error message more helpful when a user tries to upload a `.wpress` file using `/import`
* Introduced a new action type - `IMPORTS_PRE_UPLOAD_FAILED` and defined its corresponding action creator and reducing function to handle the cross-components communication between `uploading-pane` and `error-pane`. A `IMPORTS_PRE_UPLOAD_FAILED` action will be dispatched by the `uploading-pane` component when it detects the uploading file is a `.wpress` file in the `setupUpload()` function call. The `error-pane` will display the improved error message when it is notified that the error status has changed.

#### Testing Instructions

* Go to http://calypso.localhost:3000/import/{SITE_SLUG}
* Select import content from `WordPress`

Screen Shot 2022-08-22 at 3 56 58 PM

*  Click `upload it to import content`
*  Upload a `.wpress` file (Here is a [file](https://drive.google.com/file/d/1wqBy7iN5iyxJd0nRqy1OnIBj8Uwcw6iP/view?usp=sharing) for testing)
*  You should see this error message appear after the uploading is completed

Screen Shot 2022-08-24 at 8 32 51 PM

* Click the `install the plugin` link should head to `/plugins/all-in-one-wp-migration`

Screen Shot 2022-08-22 at 4 04 34 PM

* Click the `try out Everything import` link should head to `http://calypso.localhost:3000/setup/import?siteSlug={SITE_SLUG}`

Screen Shot 2022-08-22 at 3 14 05 PM

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

**Review by ouikhuan — CHANGES_REQUESTED** [review]
It looks like we're capturing and showing the failure after the export file is uploaded at the moment, but I'm wondering if this is something we can do beforehand. I'm concerned that a user may have a large `.wpress` file that they took a long time to upload, only to find out after a long wait that the file is not supported. 

Maybe this is something we can check on `setupUpload` ? 

Also, it looks like the message is the same on a simple site, but either All-in-one WP migration or Import everything can only be done on Atomic sites.

**Comment by vishnugopal** [comment]
Am ok with linking to the same All-in-One WP Importer link on Simple sites too as a Simple site customer has an option to upgrade there.

**Comment by jimmy319** [comment]
> It looks like we're capturing and showing the failure after the export file is uploaded at the moment, but I'm wondering if this is something we can do beforehand. I'm concerned that a user may have a large `.wpress` file that they took a long time to upload, only to find out after a long wait that the file is not supported.
> 
> Maybe this is something we can check on `setupUpload` ?

Thanks for Wayne's suggestion. Doing the file extension check prior to the uploading process to fail fast brings an even better experience. I've implemented the new checking logic at https://github.com/Automattic/wp-calypso/pull/66674/commits/ebe3c984a15a3ec0920b2c9667a99504d353bc44 https://github.com/Automattic/wp-calypso/pull/66674/commits/8ade8effcf7c8d913837e5c285ae6bec2bd383d0. Plus, I also updated the proposed changes of PR description.

File: client/my-sites/importer/error-pane.jsx [hunk 1]
```diff
@@ -76,17 +77,27 @@ class ImporterError extends PureComponent {
...
+		const defaultError = this.props.translate(
+			'Oops! We ran into an unexpected error while uploading your file.'
 		);
+
+		if ( this.props.code === 'WPRESS_FILE_IS_NOT_SUPPORTED' ) {
+			return this.props.translate(
+				'You have uploaded a .wpress file that works with the All-in-One WP Migration plugin. You can either {{ip}}install that plugin{{/ip}}, or {{ei}}try out Everything Import{{/ei}}. {{cs}}Still need help{{/cs}}?',
+				{
+					components: {
+						br: <br />,
```
Comment (ouikhuan) [inline client/my-sites/importer/error-pane.jsx]: Do we need this br here? It looks like there's none in the sentence above?

File: client/my-sites/importer/error-pane.jsx [hunk 2]
```diff
@@ -76,17 +77,27 @@ class ImporterError extends PureComponent {
...
-					ip: <Button className="importer__error-pane is-link" onClick={ this.installPlugin } />,
-					ei: <Button className="importer__error-pane is-link" onClick={ this.everythingImport } />,
-					cs: <Button className="importer__error-pane is-link" onClick={ this.contactSupport } />,
-				},
-			}
+		const defaultError = this.props.translate(
+			'Oops! We ran into an unexpected error while uploading your file.'
 		);
+
+		if ( this.props.code === 'WPRESS_FILE_IS_NOT_SUPPORTED' ) {
```
Comment (ouikhuan) [inline client/my-sites/importer/error-pane.jsx]: I really like the idea of introducing error code in the file! To keep it easier to manage in the future, could we change `WPRESS_FILE_IS_NOT_SUPPORTED` to a constant? It looks like `client/blocks/importer/wordpress/types.ts` might be a good place to place. Maybe adding a new enum called `WPImportError` and adding this in. What do you think?

File: client/my-sites/importer/error-pane.jsx (hunk 1 above)
Comment (jimmy319) [reply client/my-sites/importer/error-pane.jsx]: Good catch! The useless code was removed at https://github.com/Automattic/wp-calypso/pull/66674/commits/07674dba9776c28b4c5fb3fa4d69e57f631e4f06

File: client/my-sites/importer/error-pane.jsx (hunk 2 above)
Comment (jimmy319) [reply client/my-sites/importer/error-pane.jsx]: I agree with your idea. Turning the error code into a constant can make our life easier. The static analysis of TS's type system can keep the application from typo bugs. Patched at https://github.com/Automattic/wp-calypso/pull/66674/commits/4f65baaa26f97730ed44e11788974f23d2fb99fb

**Review by ouikhuan — APPROVED** [review]
LGTM, great work @jimmy319 !

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7503345

Thank you @jimmy319 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7503345).

**Pull request merged**